components/NewNFTBook.vue

```diff
@@ -680,7 +596,42 @@ useSeoMeta({
 onMounted(async () => {
   try {
     isLoading.value = true
-    await fetchStripeConnectStatusByWallet(wallet.value)
+
+    if (isEditMode.value) {
+      await fetchStripeConnectStatusByWallet(wallet.value)
+      const classResData: any = await $fetch(`${LIKE_CO_API}/likernft/book/store/${classId.value}`, {
+        headers: {
+          authorization: `Bearer ${token.value}`
+        }
+      })
+      if (classResData) {
+        shippingRates.value = classResData?.shippingRates || []
+        if (classResData?.ownerWallet !== wallet.value) {
+          throw new Error('NOT_OWNER_OF_NFT_CLASS')
+        }
+        if (classResData.prices.length > 0) {
+          const currentEdition = classResData.prices.filter(e => e.index.toString() === editionIndex.value)[0]
```

Comment:
Use `.find`

Reply:
[🎨 Enhance error handling for class and edition retrieval](https://github.com/likecoin/nft-book-press/pull/332/commits/8dc7208c0da428dc65495b986abf1fb1fe89709e)